src/Front/Parser.y

```diff
@@ -270,16 +281,12 @@ exprPar                               --> [Expr]
   | expr                                { [$1] }
 {
 -- | What to do upon encountering parse error.
---
--- TODO does this really just throw an uncaught exception on parse error, or
--- does Happy catch that for us?
-parseError :: Token -> a
-parseError (Token (sp, _)) =
-    error $ show (pretty sp) ++ ":Syntax error"
+parseError :: Token -> Alex a
+parseError (Token (sp, _)) = syntaxErr $ "near " ++ show (pretty sp)
```

Comment:
near? not "in" or something? i am not fully sure what span looks like, but i'd assume the error is somewhere inside the span? unless you are saying near just to be safe since the code looks very different the further down the compiler pipeline you go?

Reply:
the syntax werror isn't inside of span, since span just describes the extent of a single token. And it only describes the unexpected token:

```
$ stack exec sslc -- -
x =
  let y = blah blah
ParseError (ErrorMsg "near 3:1     [24+0]  ")
```

The right error message here is that there is an unfinished expression introduced by the `let` keyword, but the parse error complains about the newline at the end of the file.

I'll change this to "at" but we should have very little confidence in this generated parser at providing useful error messages.